### ISSUE TITLE
Platform stable graph partitioner

### ISSUE THREAD

**Issue by schnellerhase** [FEniCS/dolfinx#3358]
### Describe new/missing feature

During the design of parallel test cases for PR (TODO) it was observed that the graph partitioners (at least scotch and parmetis, did not test kahip) produce unstable results across different platforms. 

This is a problem when it comes to reproducibility of results, especially for test cases. Tests running on a different platform will create differently parallelized meshes and thus mesh structures. Additionally for a user it might be hard to understand why a platform change (mind you with same parallelization settings) yields different, not strictly, comparable results - this should be at least be possible to enforce.

To fix this one would either need to figure out how to make these behave equally - tried that myself without luck - or introduce some kind of nice interface to facilitate this, i.e. a partitioner dummy that reproduces a given (previously computed) partitioning.

In the before mentioned PR this is facilitated by just mimicking the behavior of `graph::scotch::partitioner(...)` explicitly and switching the result depending on the current communicator size.

**Comment by garth-wells** [comment]
I don't think this is worth the effort, and we should really design tests that are independent of the partitioning (and (re-)ordering too).

**Comment by jorgensd** [comment]
I think reading in the partitioning information from an existing mesh would resolve this for most users (https://scientificcomputing.github.io/io4dolfinx/docs/api.html#io4dolfinx.read_mesh) where you can set `read_from_partition=True` if partitioning information is stored.

**Comment by MariusCausemann** [comment]
I have had a case where differences in the partitioning had a big impact on the conditioning of the system, causing non-deterministic solver failures. For debugging purposes, I think it would be very useful to be able to reproduce the same partitioning deterministically. @jorgensd suggestion sounds like a good way to resolve this!
